src/structure.jl

```diff
@@ -0,0 +1,183 @@
+module Structure
```

Comment:
can you please create a markdown file for structure.md in docs/API to explain in a few sentences what this module is about

Reply:
I've slightly improved docs [here](https://github.com/NQCD/NQCDynamics.jl/blob/d7f16e2ea8a07ec4433d8fa66933fa23775886b0/docs/src/api/NQCDynamics/structure.md).